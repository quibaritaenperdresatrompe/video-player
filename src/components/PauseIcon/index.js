import glamorous from 'glamorous'
import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'

const StyledSvg = glamorous.svg(props => ({
  fill: props.color,
  ':hover': {
    fill: props.backgroundColor,
  },
}))

class PauseIcon extends PureComponent {
  render() {
    return (
      <StyledSvg
        backgroundColor={this.props.backgroundColor}
        color={this.props.color}
        height={this.props.size}
        viewBox='0 0 24 24'
        width={this.props.size}
      >
        <path d="M6 19h4V5H6v14zm8-14v14h4V5h-4z"/>
        <path fill="none" d="M0 0h24v24H0z"/>
      </StyledSvg>
    )
  }
}

PauseIcon.propTypes = {
  backgroundColor: PropTypes.string,
  color: PropTypes.string,
  size: PropTypes.string,
}

PauseIcon.defaultProps = {
  backgroundColor: 'hsla(0, 0%, 100%, 1)',
  color: 'hsla(0, 0%, 100%, 0.8)',
  size: '1em',
}

export default PauseIcon
