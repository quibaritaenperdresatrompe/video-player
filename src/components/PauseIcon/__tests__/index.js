import PauseIcon from '../'

import itRendersAllMutations from '../../../utils/jest-it-renders-all-mutations'

const mutations = [
  {
    name: 'without props',
    props: null,
  },
  {
    name: 'with all props',
    props: {
      backgroundColor: '#000',
      color: '#fff',
      size: '3em',
    },
  },
]

describe(PauseIcon.name, () => {
  itRendersAllMutations(PauseIcon, mutations)
})
