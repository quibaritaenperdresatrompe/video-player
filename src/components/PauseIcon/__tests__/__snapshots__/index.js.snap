// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PauseIcon with all props it renders correctly 1`] = `
<svg
  className="css-1hgjwxk"
  color="#fff"
  height="3em"
  viewBox="0 0 24 24"
  width="3em"
>
  <path
    d="M6 19h4V5H6v14zm8-14v14h4V5h-4z"
  />
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
</svg>
`;

exports[`PauseIcon without props it renders correctly 1`] = `
<svg
  className="css-189tqkb"
  color="#fff"
  height="1em"
  viewBox="0 0 24 24"
  width="1em"
>
  <path
    d="M6 19h4V5H6v14zm8-14v14h4V5h-4z"
  />
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
</svg>
`;
